docs/modules/rmt.md

```diff
@@ -0,0 +1,126 @@
+```lua
+```
+
+#### Options table
+
+This optional table consists of a number of keys that control various aspects of the RMT transmission.
+
+- `invert` if true, then the output is inverted.
+- `carrier_hz` specifies that the signal is to module the carrier at the specified frequency. This is useful for IR transmissions.
+- `carrier_duty` specifies the duty cycle of the carrier. Defaults to 50%
+- `idle_level` specifies what value to send when the transmission completes.
+
+## rmt.rxsetup(gpio, bitrate, options)
+
+This sets up a receive channel on the specified gpio pin at the specified rate. Various options described below
+can be specified in the `options`. The bit time is specified in picoseconds so that integer values can be used.
+
+An error will be thrown if the bit time cannot be approximated.
+
+#### Syntax
+`channel = rmt.rxsetup(gpio, bittime[, options])`
+
+#### Parameters
+- `gpio` The GPIO pin number to use.
+- `bittime` The bit time to use in picoseconds. Only certain times can be handled exactly. The actual set time will be returned. The actual range is limited -- probably using 100,000 (0.1&micro;S) or 1,000,000 (1&micro;S).
+- `options` A table with the keys as defined below.
+
+##### Returns
+- The `rmt.channel` object that can be used for receiving data
+- The actual bit time in picoseconds.
+
+#### Example
+```lua
+```
+
+#### Options table
+
+This optional table consists of a number of keys that control various aspects of the RMT transmission.
+
+- `invert` if true, then the input is inverted.
+- `filter_ticks` If specified, then any pulse shorter than this will be ignored.
+- `idle_threshold` If specified, then any level longer than this will set the receiver as idle. The default is 65535 bit times.
+
+
+## channel:on(method, callback)
+
+This is establishes a callback to use when data is received and it also starts the data reception process. It can only be called once per receive
+channel.
+
+#### Syntax
+`channel:on(event, callback)`
+
+#### Parameters
+- `event` This must be the string 'data' and it sets the callback that gets invoked when data is received.
+- `callback` This is invoked with a single argument that is a string that contains the data received in the format described for transmit above. `struct.unpack` is your friend.
+
+#### Returns
+`nil`
+
+## channel:send(data, cb)
+
+This is a (default) blocking call that transmits the data using the parameters specified on the `txsetup` call.
+
+#### Syntax
+`channel:send(data[, cb])`
+
+#### Parameters
+- `data` This is either a string or a table of strings.
+- `cb` THis is an optional callback when the transmission is actually complete. If specified, then the `send` call is non-blocking, and the callback invoked when the transmission is complete. Otherwise the `send` call is synchronous and does not return until transmission is complete.
+
+#### Data Encoding
+
+If the `data` supplied is a table, then the elements of the table are concatenated together and sent.
```

Comment:
Table as in "array-like"? Or random key/value tables will also work (and is so, what order)?